pkg/client/fake/client_test.go

```diff
@@ -541,6 +543,32 @@ var _ = Describe("Fake client", func() {
 			}
 		})
 
+		It("should be able to watch", func() {
+			By("Creating a watch")
+			objWatch, err := cl.Watch(context.Background(), &corev1.ServiceList{})
+			Expect(err).NotTo(HaveOccurred())
+
+			defer objWatch.Stop()
+
+			go func() {
+				defer GinkgoRecover()
+				// It is likely starting a new goroutine is slower than progressing
+				// in the outer routine, sleep to make sure this is always true
+				time.Sleep(100 * time.Millisecond)
```

Comment:
Why not a channel or semaphore to enforce it?

Reply:
What we are waiting for is that the main routine starts to read, how would you sync that with a channel or semaphore?